### PR TITLE
Try fixing build

### PR DESCRIPTION
**Pull request by pwalczysko**
After some attempts at [prod-playbooks](https://github.com/ome/prod-playbooks/pull/370) trying to fix here as this is a simpler repo.

cc @sbesson @jburel

### CODE REVIEW

**Comment by pwalczysko** [comment]
Very similar expereince to prod-playbooks repo fixing - the https://github.com/ome/prod-playbooks/pull/370#issuecomment-1573594780 repeats here too.

**Comment by sbesson** [comment]
Looking at the error, I think you'll need https://github.com/IDR/deployment/commit/03cf8347e717201e89f5777c80210d68f06c643e - see https://github.com/ansible-community/molecule/issues/3903

**Comment by pwalczysko** [comment]
Thanks @sbesson , this fixed the error and I have some progress. Next step is failing with molecule not finding the roles.
I think this is https://github.com/ansible-community/molecule/issues/3404 - but would you please have an insight how would I implement the fix https://github.com/ansible-community/molecule/discussions/3569 on my side ?

**Comment by sbesson** [comment]
Try the equivalent of https://github.com/ome/ansible-role-ssl-certificate/commit/10d5013ade3f3c9f22b716a3b4860deac57a8207 i.e. use the FQDN role name in the Molecule playbooks?

**Comment by pwalczysko** [comment]
Atm stuck at 

```
the role 'ome.omero_common' was not found in ansible.legacy:/home/runner/work/ansible-role-omero-web/ansible-role-omero-web/molecule/resources/roles:/home/runner/.cache/ansible-compat/ae077f/roles:/home/runner/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/home/runner/work/ansible-role-omero-web:/home/runner/work/ansible-role-omero-web/ansible-role-omero-web/molecule/resources```

it seems that https://github.com/ome/ansible-role-omero-web/pull/45/commits/6fe2254622ef4291960674f559c39227d54c6edc fixed the similar errror of missing role complaint for ``ome.omero_web`` as per @sbesson's hint, but for this ^^^ ``ome.omero_common`` the only idea I have is https://github.com/ansible/ansible-lint/issues/1387 - but again, lost in the implementation of solution. The [pinning of the ansible-lint](https://github.com/ansible/ansible-lint/issues/1387#issuecomment-788168807) did not do it, and for all I can gather, the problem shoiuld not be there in newest ansible-lint...

Maybe the way forward would be for me to install from scratch https://molecule.readthedocs.io/installation/#pip and learn how to test the minimal set of scenarios ?

**Comment by sbesson** [comment]
Confusingly, I fear the error you have been trying to fix is not a fatal one.  The same message is thrown in https://github.com/ome/ansible-role-postgresql/actions/runs/4446829141/jobs/7807628554  which is passing.

I think the critical error comes from the lint step above

```
Molecule active-ubuntu2004 > lint
./.github/workflows/molecule.yml
  Error: 38:81 [line-length] line too long (83 > 80 characters)
 ```

It might be worth fixing the dependency error in all cases to reduce the noise. Looking at the logs, the issue is that the role gets installed in the `dependency` phase under a folder named after the scenario

```
Molecule active-ubuntu2004 > dependency
  INFO     Running from /home/runner/work/ansible-role-omero-web/ansible-role-omero-web : ansible-galaxy collection install -vvv community.docker:>=3.0.2
  INFO     Running from /home/runner/work/ansible-role-omero-web/ansible-role-omero-web : ansible-galaxy collection install -vvv ansible.posix:>=1.4.0
  Starting galaxy role install process
  - downloading role 'omero_common', owned by ome
  - downloading role from https://github.com/ome/ansible-role-omero-common/archive/0.3.4.tar.gz
  - extracting ome.omero_common to /home/runner/.cache/molecule/ansible-role-omero-web/active-ubuntu[20](https://github.com/ome/ansible-role-omero-web/actions/runs/5178132773/jobs/9329184347?pr=45#step:5:21)04/roles/ome.omero_common
  - ome.omero_common (0.3.4) was installed successfully
```

but the lint step above looks into 

```
syntax-check[specific]: the role 'ome.omero_common' was not found in ansible.legacy:/home/runner/work/ansible-role-omero-web/ansible-role-omero-web/molecule/resources/roles:/home/runner/.cache/ansible-compat/ae077f/roles:/home/runner/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/home/runner/work/ansible-role-omero-web:/home/runner/work/ansible-role-omero-web/ansible-role-omero-web/molecule/resources
```
I have not checked what the best practice is

**Comment by pwalczysko** [comment]
I think now I am failing with 

```
fatal: [omero-web-active-centos7]: FAILED! => {"changed": false, "cmd": "/bin/systemctl", "msg": "Failed to get D-Bus connection: No such file or directory", "rc": 1, "stderr": "Failed to get D-Bus connection: No such file or directory\n", "stderr_lines": ["Failed to get D-Bus connection: No such file or directory"], "stdout": "", "stdout_lines": []}
```

**Comment by pwalczysko** [comment]
Well, but maybe I am failing with 

```
 Error: the role 'ome.omero_common' was not found in ansible.legacy:/home/runner/work/ansible-role-omero-web/ansible-role-omero-web/molecule/resources/roles:/home/runner/.cache/ansible-compat/ae077f/roles:/home/runner/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/home/runner/work/ansible-role-omero-web:/home/runner/work/ansible-role-omero-web/ansible-role-omero-web/molecule/resources
``` 

now ? I can see those two errors (the one above in this comment and the one in https://github.com/ome/ansible-role-omero-web/pull/45#issuecomment-1578341478 - but which one is the failing one is not clear

**Comment by pwalczysko** [comment]
All jobs in the buid are green now. cc @sbesson @jburel

**Comment by pwalczysko** [comment]
In summary, the level of testing is as it was before this PR. Did not have to remove any tests. Will see if similar tactics will work also with the other ansible repos (not probable) - but for the moment, I do not see a reason for announcing the deprecation of testing, at least not on this repo @sbesson

File: molecule/active-ubuntu2004/Dockerfile.j2 [hunk 1]
```diff
@@ -6,7 +6,7 @@ FROM {{ item.registry.url }}/{{ item.image }}
 FROM {{ item.image }}
 {% endif %}
 
-RUN if [ $(command -v apt-get) ]; then apt-get update && apt-get install -y python sudo bash ca-certificates && apt-get clean; \
+RUN if [ $(command -v apt-get) ]; then apt-get update && apt-get install -y python sudo bash ca-certificates gnupg && apt-get clean; \
```
Comment (sbesson) [inline molecule/active-ubuntu2004/Dockerfile.j2:9]: Should this be moved to a `prepare.yml` playbook?

File: molecule/active-ubuntu2004/molecule.yml [hunk 2]
```diff
@@ -16,6 +18,10 @@ platforms:
       - /sys/fs/cgroup
 provisioner:
   name: ansible
+  config_options:
```
Comment (sbesson) [inline molecule/active-ubuntu2004/molecule.yml:21]: Can you add a comment explaining why this is required so that we can come back to it later?

**Review by sbesson — CHANGES_REQUESTED** [review]
A few inline questions. A more general one is whether the Ubuntu 18.04 scenario should be kept at all since this OS is EOL and there is really no driver to keep supporting it to the best of my knowledge

File: molecule/active-ubuntu2004/molecule.yml (hunk 2 above)
Comment (pwalczysko) [reply molecule/active-ubuntu2004/molecule.yml:21]: I will remove, this was added in attempt to fix a problem which somehow `went away by itself`.

File: molecule/active-ubuntu2004/Dockerfile.j2 (hunk 1 above)
Comment (pwalczysko) [reply molecule/active-ubuntu2004/Dockerfile.j2:9]: I can try to move the gnupg. In my tries, the success of installing something via the ``prepare.yml`` playbook was Zero. The complaint was always ``omero-web-active-ubuntu2004]: FAILED! => {"changed": false, "msg": "No package matching 'xxx' is available"} . Replace xxx with `curl`, gnupg etc as appropriate.

Comment (pwalczysko) [reply molecule/active-ubuntu2004/Dockerfile.j2]: This is fixed as far as I can with https://github.com/ome/ansible-role-omero-web/pull/45/commits/65e2b70461f7740539ea06e6586aa4df17a8ff85

The reasoning is:

1. Tried to remove the Dockerfile completely and put the packages installs into the [prepare.yml](https://github.com/ome/ansible-role-omero-web/pull/45/commits/65e2b70461f7740539ea06e6586aa4df17a8ff85#diff-b5f60a491d9c8f17eb774de1a16d0d8eec5bc6f14e25355e910212fb5b11a4c9). This failed with subsequent report that pkg such-and-such (in the example below it is ``curl``, the first in the list) cannot be installed:

https://github.com/ome/ansible-role-omero-web/actions/runs/5254245234/jobs/9492614696

``no package matching curl is available``

2. Returned the Dockerfile back, but deleted all the package installs from it. This was working, except for sudo, which failed with


https://github.com/ome/ansible-role-omero-web/actions/runs/5254389653/jobs/9492935545:
```
TASK [Install packages] ********************************************************
  fatal: [omero-web-active-ubuntu2004]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: 1: sudo: not found\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 127}

File: molecule/active-ubuntu2004/molecule.yml (hunk 2 above)
Comment (pwalczysko) [reply molecule/active-ubuntu2004/molecule.yml]: fixed in https://github.com/ome/ansible-role-omero-web/pull/45/commits/b8d0468564388e0ad9c75fd810dc4f774fa6c793

**Comment by pwalczysko** [comment]
> A few inline questions. A more general one is whether the Ubuntu 18.04 scenario should be kept at all since this OS is EOL and there is really no driver to keep supporting it to the best of my knowledge

Fixed in https://github.com/ome/ansible-role-omero-web/pull/45/commits/483c38581f378e740eded8b3a17e5791df0ad4e7

**Review by sbesson — APPROVED** [review]
Obviously lots of commits here which capture the trial-and-error nature of the investigation. Overall the final set of changes are fairly minimal, the scenario for the EOL Ubuntu 18.04 distribution is removed and the changes are inline with Molecule 5.

I am still uncertain about the best way to handle the testing dependencies. This PR inherits some of the proposals made in https://github.com/ome/ansible-role-postgresql/pull/32 to remove `ome/action-ome-ansible-molecule` which installs `ome-ansible-molecule` in favor of a more direct definition within the GitHub workflow itself. This offers more control when looking into a large upgrade like this one but at the cost of losing the unification between all roles as each of them can now pin its own requirements. I suspect more discussions and decisions will need to happen with regards to the mid-term maintenance of this infrastructure.

Immediately, this fixes the CI build and I have no objection to getting this merged under these terms

**Comment by pwalczysko** [comment]
@jburel Do you want to merge ?

**Pull request merged**